### PR TITLE
Update license from EPL v1.0 to EPL 2.0

### PR DESCRIPTION
**Pull request by charliegracie**
Update the copyright headers, license files, notices, etc based on the voting in #1596 and the agreement by the contributors in #1606.

This single PR is a combination of #1617, #1618, #1622, #1625 and #1627 and 2 other commits to complete the update.

### CODE REVIEW

**Comment by charliegracie** [comment]
I created this PR so that all of the license changes could be released at one time.  I think it would be beneficial to have a single commit that changes the copyright header, license files, notices, etc.  The separate commits were useful in reviewing the changes but I think squashing them when they are merged would be preferable so that not matter which SHA is checked out from the git repo you have a complete view of the licensing at that point.

@DanHeidinga and @keithc-ca are you ok with this PR consuming the changes from the other PRs?

@mstoodle what are your thoughts?

**Comment by DanHeidinga** [comment]
> are you ok with this PR consuming the changes from the other PRs?

+1 from me.  Thanks for merging this down to a single PR.

**Comment by mstoodle** [comment]
Thanks @charliegracie ; I *really* think this is the best approach.

**Comment by keithc-ca** [comment]
+1 We missed configure and CONTRIBUTING.md: thanks @charliegracie for catching those files.

File: README.md [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+OpenJDK Assembly Exception [2].
 
-    The Apache License v2.0 is available at
-    http://www.opensource.org/licenses/apache2.0.php
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
 
-Contributors:
-  Multiple authors (IBM Corp.) - initial implementation and documentation
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
```
Comment (mstoodle) [inline README.md]: There is another section lower down that still references the older license (lines 93-95: What's the License?)

File: ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
- *      http://www.eclipse.org/legal/epl-v10.html
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
  *
- *      The Apache License v2.0 is available at
- *      http://www.opensource.org/licenses/apache2.0.php
- *
- * Contributors:
- *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
```
Comment (mstoodle) [inline ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp]: there's a busted comment at line 157 in this file that happens to reference the epl-v10 file name. It was there before this PR and the comment has nothing to do with our licensing, just pointing it.

Comment (charliegracie) [reply ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp]: Sadly this was in the original version of the file.  I am not sure what it is supposed to be.  I will reach out to the developer to see if they have any input but at this point I am going to ignore if for this PR

Comment (keithc-ca) [reply ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp]: @jason-hall Can you weigh in on what that comment should say? Is the reference to epl-v10 significant?

**Comment by mstoodle** [comment]
LGTM, will wait for tests to complete just to be on the safe side.

**Comment by mstoodle** [comment]
Thanks @charliegracie @DanHeidinga @keithc-ca and anyone else I missed for your efforts to make this happen as an atomic commit.

Merging now.

**Comment by mstoodle** [comment]
Ok, apparently I merged without squashing. That was the plan, but apparently that's not the current reality.

*sigh* good thing I didn't thank myself up there for any effort to make it an atomic commit.

I cannot fix it now without rewriting history in our git repository :(

**Pull request merged**